bridges/KilledbyGoogleBridge.php

```diff
@@ -10,9 +10,10 @@ class KilledbyGoogleBridge extends BridgeAbstract {
 
 	public function collectData() {
 
-		$json = getContents(self::URI . '/graveyard.json')
-			or returnServerError('Could not request: ' . self::URI . '/graveyard.json');
```

Comment:
Huh, how did this ever work? It looks like `or`, unlike in JavaScript, seems to cast the result to `bool`:

```
php > var_dump(["test"] or (function() {})());
bool(true)
```

So we would need to use something like `?:`. Then the `returnServerError` could make sense as a replacement for `throw`, which [can only be used in expressions since PHP 8.0](https://php.watch/versions/8.0/throw-expressions).

But I agree that moving it to is statement is cleaner.

I do not have an opinion on distinguishing `Client` vs. `Server` error but we should at least have a custom `Exception` subclass (e.g. `DataCollectionException`).

Reply:
I think `or` is the same as `||` only that it has a lower precedence. I'm not sure how much lower. It's clearly confusing.

Reply:
Yeah, that is correct, according to my understanding. And the precedence would be irrelevant either way, since function calls bind the tightest (not even listed in the [precedence table](https://www.php.net/manual/en/language.operators.precedence.php)). So either the original author wanted to use `or` for the nice English-like form, or they were just confused.